### ISSUE TITLE
[Feature] actions/getters/mutations

### ISSUE THREAD

**Issue by HerringtonDarkholme** [HerringtonDarkholme/kilimanjaro#5]
These method can define multiple items in kilimanjaro.

```ts
store.getters(p => ({
  coffee: p(state => state.cappuccino),
  remainingAnko: p(state => state.ankoAmount - state.matchaAmount)
}))
.mutations(p => ({
  payCheck: p(state => () => state.order += 1),
  eat_sweet: p(state => n => state.ankoAmount -= n)
}))
```

This requires mapping types, index access type and `keyof` typings. 
The `p` place holder is mainly for type checking. Because TypeScript cannot index accessed type for return type.
Also requires index access type on type constraints.

**Comment by HerringtonDarkholme** [comment]
isomorphic mapped types can enable this feature

```ts
store.getters<P>(config: {[K in keyof P]: (s: state) => P[K]})
```

**Comment by HerringtonDarkholme** [comment]
Very hard to implement since overloading choosing is too finicky in TypeScript. 😞 
Need to split two API

**Comment by HerringtonDarkholme** [comment]
Shipped in 0.6.0
